### PR TITLE
Fix findAllRefs, getHighlightSpans, renameLocs, renameInfo for default exports and functions expressions

### LINKED ISSUE #3065: Can't Refactor -> Rename a default export name in Visual Studio
Hi. I'm using VS 2013 and TypeScript 1.5 Beta. Amazing work, as always.

I can't seem to rename default exports via "Rename". Consider this:

```TypeScript
export default class InfoPanelViewModel {
}
```

- Invoke the Rename operation 
- Punch in a new name, submit via click or return
- Notice no change in the symbol's name

Doing a Refactor -> Rename does nothing. Tested with `function`, seems to be for anything `export default`. Removing the `default` keyword allows me to rename the symbol. Adding it back prevents it again.

Is this by design, or a bug? If it's by design, perhaps a warning or error would be useful.

### LINKED ISSUE #3230: Named function expressions get incorrect display names when renaming
image

Note that the dialog box shows "Rename '__function'" instead of "Rename 'foo'"

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
* Fixes #3065
* Fixes #3230

Tip: Just review the changed files.

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -4941,73 +4941,37 @@ module ts {
...
                 if (isImportOrExportSpecifierName(location)) {
                     return location.getText();
                 }
 
                 // Special case for function expressions, whose names are solely local to their bodies.
-                let functionExpression = forEach(declarations, d => d.kind === SyntaxKind.FunctionExpression ? <FunctionExpression>d : undefined);
+                // TODO (drosen): Why would we have to lookup the interned name for a function expression?
+                //                Shouldn't we have found a scope? Consider a 'Debug.fail()'.
+                let functionExpression: FunctionExpression;
+                if (symbol.valueDeclaration && symbol.valueDeclaration.kind === SyntaxKind.FunctionExpression) {
```
Comment (mhegazy) [inline src/services/services.ts]: move this to getSymbolScope

File: src/services/utilities.ts [hunk 2]
```diff
@@ -656,4 +656,34 @@ module ts {
     export function isJavaScript(fileName: string) {
         return fileExtensionIs(fileName, ".js");
     }
+
+    export function getDeclaredName(typeChecker: TypeChecker, symbol: Symbol, location: Node) {
+        // If this is an export or import specifier it could have been renamed using the 'as' syntax.
+        // If so we want to search for whatever is under the cursor.
+        if (isImportOrExportSpecifierName(location)) {
```
Comment (mhegazy) [inline src/services/utilities.ts]: as per our discussion we need to remove this, and put it in getSymbolInfo instead. and this way getInternedName is just symbol.name all the time and we can remove it.

**Comment by DanielRosenwasser** [comment]
We're not taking this in for 1.5.

**Comment by mhegazy** [comment]
we need to merge this into master though.

**Comment by DanielRosenwasser** [comment]
Yup rebasing now.

**Pull request closed without merging**